### ISSUE TITLE
User can type anything in the text boxes and crash the editor

### ISSUE THREAD

**Issue by T1ti** [stoneharry/WoW-Spell-Editor#151]
**Spell editor version**
1.9.9

**Describe the bug**
The user can type anything in the text box instead of an expected integer/float
![image](https://user-images.githubusercontent.com/40864460/139757062-db9e82b2-24c5-40da-b252-9ff5eac92655.png)
It crashes when saving because spell editor expects an integer/float
![image](https://user-images.githubusercontent.com/40864460/139757085-51ee49da-8321-44c1-a0e8-35594643d0aa.png)

**To Reproduce**
1. Go to any text box that should have a number
2. type anything with letter of symbols
4. See error when saving

**Expected behavior**
Don't allow the user to type anything but numbers ? (and a digit " . " )

**Comment by stoneharry** [comment]
I don't think it crashes. It just displays a popup message that can be dismissed. Is this really an issue?

**Comment by T1ti** [comment]
> I don't think it crashes. It just displays a popup message that can be dismissed. Is this really an issue?

it is, becausse if the user accidentally type something in like the level box and then moves on to editing effects, when he tries to save it just errors and he probably will not figure out why as the error message is not very readable

Btw I think most of those numbers only text boxes should be replaced by spinboxes, they're more convenient for numbers
